### PR TITLE
docs(JSStylguide): Misc changes (FRMW-19)

### PR DESCRIPTION
**Pull request by parlarjb**
Changes for https://jira.rax.io/browse/FRMW-19

### CODE REVIEW

**Comment by TheNando** [comment]
LGTM

File: guides/js-styleguide.md [hunk 1]
```diff
@@ -58,16 +58,16 @@ if (condition && otherCondition) {
```
Comment (CITguy) [inline guides/js-styleguide.md]: Is the Module Definitions section still relevant considering that our prerequisite for new components use the component scaffolding (https://github.com/rackerlabs/encore-ui/blob/master/CONTRIBUTING.md#pr-steps)?

File: guides/js-styleguide.md [hunk 2]
```diff
@@ -84,11 +84,11 @@ For any new functionality that can't be tested via unit tests (or is outside of
...
 
-**For more on both unit and midway testing, read [the EncoreUI Testing Handbook](./testing.md).**
+**For more on both unit and midway testing, read [the EncoreUI Testing documentation](./testing.md).**
 
 # Other Guidelines
 
-* [Write Maintainble Code](http://www.youtube.com/watch?v=c-kav7Tf834)
+* [Write Maintainable Code](http://www.youtube.com/watch?v=c-kav7Tf834)
 * The less code on one line, the easier it is to read and the less likely you’ll encounter a merge conflict.
 * Adding `'use strict';` to your files isn't necessary, as JSHint and this styleguide take care of most everything that `use strict` helps with. As a rule, we won't
```
Comment (CITguy) [inline guides/js-styleguide.md]: This entry seems to be incomplete. `As a rule, we won't` ...

File: guides/js-styleguide.md (hunk 1 above)
Comment (parlarjb) [inline guides/js-styleguide.md]: I mention in this section that the component scaffolding should be used, and I think it's worthwhile to show how we set up our module system.

**Comment by parlarjb** [comment]
Updated.

**Comment by CITguy** [comment]
LGTM

**Pull request merged**